tapdb/asset_minting.go

```diff
@@ -328,6 +328,37 @@ func (a *AssetMintingStore) CommitMintingBatch(ctx context.Context,
 			}
 		}
 
+		// If the batch is funded, we can also insert the batch TX and
+		// batch genesis outpoint.
+		if newBatch.GenesisPacket != nil {
+			genesisPacket := newBatch.GenesisPacket
+			genesisTx := genesisPacket.Pkt.UnsignedTx
+			changeIdx := genesisPacket.ChangeOutputIndex
+			genesisOutpoint := genesisTx.TxIn[0].PreviousOutPoint
+
+			var psbtBuf bytes.Buffer
+			err := genesisPacket.Pkt.Serialize(&psbtBuf)
+			if err != nil {
+				return fmt.Errorf("unable to encode psbt: "+
+					"%w", err)
+			}
+
+			genesisPointID, err := upsertGenesisPoint(
+				ctx, q, genesisOutpoint,
+			)
+			if err != nil {
+				return fmt.Errorf("unable to upsert genesis "+
+					"point: %w", err)
+			}
+
+			return q.BindMintingBatchWithTx(ctx, BatchChainUpdate{
```

Comment:
Passing thought: does this invalidate any existing logic that assumes that only a near finalized batch can have a batch tx bound to it?

Reply:
Good Q, so far my conclusion was no, as the funding happened in the first state of the caretaker and that field of the batch struct isn't modified before then. Since we don't unwind batch funding right now (#830), there's no logic around that to change.

We may want to delete genesis points from cancelled batches later on.